### DISCUSSION TITLE
Overriding Dates

### DISCUSSION THREAD

**Discussion by DellanX** [spatie/icalendar-generator#109] [Ideas]
Needs support for creating EXDATES. Even if it was creating a secondary event and flagging its RECURRENCE-ID to populate the original with overrides.

**Comment by rubenvanassche** [comment]
We have `doNotRepeatOn` which generates EXDATES. If something is missing in our implementation, you're always welcome to make a PR adding such feature!

**Reply by DellanX** [reply]
Actually, the functionality I am looking for is to change a specific event. Such as its date, description, or other fields.
Not on the entire series, but on the specific event by recurrence-id, further, not just deleting the date altogether

Can look into making a PR for this when work calms down a bit.

**Reply by rubenvanassche** [reply]
👍

**Reply by DellanX** [reply]
I've long since learned that the proper way to handle overriding repeating events is to create an event with the following attributes:
1. UID Duplicate to the original event's UID
2. Recurrence-ID as the datetime of the recurrrence that is being overwritten.

I don't see a way to specify Recurrence ID in this library, I figure I'll make my PR just as a method to add this property to the event?

**Reply by rubenvanassche** [reply]
Yeah, send in a PR I think these can help you get started: https://icalendar.org/iCalendar-RFC-5545/3-8-4-4-recurrence-id.html
